spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/filter/WebsocketRoutingFilter.java

```diff
@@ -94,6 +101,24 @@ public int getOrder() {
 		return filters;
 	}
 
+	private void changeSchemeIfIsWebSocketUpgrade(ServerWebExchange exchange) {
+		// Check the Upgradle
+		URI requestUrl = exchange.getRequiredAttribute(GATEWAY_REQUEST_URL_ATTR);
+		String scheme = requestUrl.getScheme();
+		String upgrade = exchange.getRequest().getHeaders().getUpgrade();
+		// change the scheme if the socket client send a "http" or "https"
+		if ("WebSocket".equalsIgnoreCase(upgrade) && ("http".equals(scheme) && !"https".equals(scheme))) {
```

Comment:
This place should be `("http".equals(scheme) || "https".equals(scheme))`?

Reply:
oh!I'm so stupid